presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergPageSource.java

```diff
@@ -211,6 +215,15 @@ private static Object deserializePartitionValue(Type type, String valueString, S
             if (type.equals(DATE)) {
                 return parseLong(valueString);
             }
+            if (type.equals(TIME)) {
+                return parseLong(valueString);
+            }
+            if (type.equals(TIMESTAMP)) {
+                return parseLong(valueString);
+            }
```

Comment:
merge the three if for DATE/TIME/TIMESTAMP?

Reply:
Thanks for the comment! I merged them into one if statement.